### ISSUE TITLE
Old unification has trouble with cumulative inductive types

### ISSUE THREAD

**Issue by SkySkimmer** [coq/coq#6976]
```coq
Set Printing Universes.

Set Universe Polymorphism.
Set Polymorphic Inductive Cumulativity.

Record Unit := tt {}.

Record Foo (A B : Type) := mkfoo { }.

Parameter V : Type -> Type.
Axiom foobar : forall A, V A -> Foo@{_ Set} A Unit.

Section sec.
  Variable A : Type.
  Variable v : V A.

  Definition bar@{i} : Foo@{i i} A Unit.
  Proof.
    Fail apply foobar.
    refine (foobar _ _). apply v.
  Qed.
End sec.
```
(originally `Foo` is `Equiv` and `V` is `Contr`)

It wants to unify `Foo@{x Set}` and `Foo@{i i}`, before doing that it checks that they have the same type. They don't so it fails.

I don't know if there's a way to make it work, maybe we just have to wait for unifall.

**Comment by ejgallego** [comment]
> I don't know if there's a way to make it work, maybe we just have to wait for unifall.

Quite a few developments have their own `apply` version anyway for this reason, so yeah, I guess people can use any of the available tactics that do work [including `apply:`]

**Comment by ppedrot** [comment]
This particular instance was solved by #14758. Let's close it even if unification is still quite oblivious.
